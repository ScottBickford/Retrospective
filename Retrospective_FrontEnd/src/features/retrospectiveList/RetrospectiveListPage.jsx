import React, { useState, useEffect, useReducer, Fragment } from 'react';
import {
  retrospectiveListReducer,
  getRetrospectives
} from './retrospectiveListActions';
import RetrospectiveList from './RetrospectiveList';
import DatePicker from 'react-datepicker';
import 'react-datepicker/dist/react-datepicker.css';

const initialData = {
  list: [],
  error: null
};

const RetrospectiveListPage = ({history}) => {
  const [data, dispatch] = useReducer(retrospectiveListReducer, initialData);
  const [filterDate, setFilterDate] = useState(null);

  useEffect(() => {
    getRetrospectives(filterDate, dispatch);
  }, [filterDate]);

  const handleAddRetrospective =() => {
    history.push('/add');
  }

  return (
    <Fragment>
      <h2>Retrospectives</h2>
      {data.error && (
        <div className='error'>An error occured: {data.error.data}</div>
      )}
      {!data.error && (
        <Fragment>
          <div style={{ paddingBottom: 10 }}>
            Filter:{' '}
            <DatePicker
              dateFormat='d MMMM yyyy'
              selected={filterDate}
              onChange={setFilterDate}
              showYearDropdown={true}
              showMonthDropdown={true}
              dropdownMode='select'
            />
            {filterDate !== null && (
              <button onClick={() => setFilterDate(null)}>Clear Filter</button>
            )}
          </div>
          <RetrospectiveList data={data} />          
          <button onClick={handleAddRetrospective} style={{marginTop:10}}>Add Retrospective</button>
        </Fragment>
      )}
    </Fragment>
  );
};

export default RetrospectiveListPage;
