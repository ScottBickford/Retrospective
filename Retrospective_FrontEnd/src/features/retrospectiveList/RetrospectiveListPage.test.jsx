import React from 'react';
import renderer from 'react-test-renderer';
import RetrospectiveListPage from './RetrospectiveListPage';

describe('RetrospectiveListPage', () => {
  test('snapshot renders', () => {    
    const component = renderer.create(<RetrospectiveListPage />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});