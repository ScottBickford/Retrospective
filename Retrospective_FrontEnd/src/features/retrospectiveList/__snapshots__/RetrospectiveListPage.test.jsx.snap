// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RetrospectiveListPage snapshot renders 1`] = `
Array [
  <h2>
    Retrospectives
  </h2>,
  <div
    style={
      Object {
        "paddingBottom": 10,
      }
    }
  >
    Filter:
     
    <div
      className="react-datepicker-wrapper"
    >
      <div
        className="react-datepicker__input-container"
      >
        <input
          className=""
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          readOnly={false}
          type="text"
          value=""
        />
      </div>
    </div>
  </div>,
  <div>
    There are no items
  </div>,
  <button
    onClick={[Function]}
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    Add Retrospective
  </button>,
]
`;
